### PR TITLE
ITN internal log GraphQL

### LINKED ISSUE #13064: Pull internal logs via control Graphql
Internal logs are used for VS' dashboard and take the following form in Ocaml codebase:

```
[%log internal] "Update_ledger_and_get_statements" ~metadata:[ ("partition", `String "single") ]
```

We need to come up with:

* Control Graphql query that would allow to pull logs starting from some index
* Control Graphql mutation that would allow to clean logs up until some index

Logs are to be stored in a bounded queue with old logs to be overwritten if not queried before that. "Bound" is to be configured via application config/CLI flag.

Feature is to be injected via setting up logging with a new backend when the `-itn-keys`/`--itn-port` flags are provided.

### PR DESCRIPTION
**Pull request by psteckler**
For the incentivized testnet GraphQL server, add query `internalLogs` and mutation `flushInternalLogs`. 

The ITN logger maintains a bounded queue of log items, consisting of a sequence number, the log message, and the log metadata. By default, the queue bound is 500; the flag `--itn-max-logs` can be used to change that.

The query takes an argument `startLogId`, the starting sequence number for the logs to be returned.

The mutation takes an argument `endLogId`, the ending sequence number for the logs to be removed from the queue. It returns a string `Deleted NN logs`.

Closes #13064.

Sample output from the query:
```
{
  "data": {
    "internalLogs": [
      {
        "id": 50,
        "message": "Update_frontier_extension",
        "timestamp": "2023-04-25 16:41:09.242914Z",
        "metadata": [
          {
            "item": "extension_name",
            "value": "ledger_table"
          }
        ]
      },
      {
        "id": 51,
        "message": "Update_frontier_extension",
        "timestamp": "2023-04-25 16:41:09.242914Z",
        "metadata": [
          {
            "item": "extension_name",
            "value": "transition_registry"
          }
        ]
      },
      {
        "id": 52,
        "message": "Update_frontier_extension",
        "timestamp": "2023-04-25 16:41:09.242914Z",
        "metadata": [
          {
            "item": "extension_name",
            "value": "best_tip_diff"
          }
        ]
      },
      {
        "id": 53,
        "message": "Update_frontier_extension",
        "timestamp": "2023-04-25 16:41:09.242914Z",
        "metadata": [
          {
            "item": "extension_name",
            "value": "snark_pool_refcount"
          }
        ]
      }
    ]
  }
}
```

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Review by MartinMinkov — APPROVED** [review]

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

File: src/lib/mina_compile_config/dune [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (library
  (name mina_compile_config)
  (public_name mina_compile_config)
- (libraries currency)
+ (libraries)
  (preprocessor_deps ../../config.mlh)
```
Comment (ejMina226) [inline src/lib/mina_compile_config/dune:5]: I might be misreading this, but if the libraries stanza is empty without the currency module, then do we even need it. Can it just be omitted?

Comment (psteckler) [reply src/lib/mina_compile_config/dune:5]: I believe the stanza could be omitted, but I kind of like having it, to demonstrate that there are no dependencies.

**Comment by psteckler** [comment]
!ci-build-me

**Review by georgeee — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Comment by psteckler** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**